packages/ui/helpers/API/mutation.ts

```diff
@@ -2,6 +2,25 @@ import { useMutation, useQueryClient } from 'react-query'
 import axios from 'axios'
 import { notification } from 'antd'
 
+function updateProgramTask(program, task) {
+  for (let i = 0; i < program.projects.length; i++) {
+    const project = program.projects[i]
+    for (let j = 0; j < project.tasks.length; j++) {
+      const task1 = project.tasks[j]
+      if (task1.taskId === task.taskId)
+        program.projects[i].tasks[j] = { ...task }
+    }
+  }
+}
+
+export function deleteProgramTask(program, task) {
+  for (let i = 0; i < program.projects.length; i++) {
+    const project = program.projects[i]
+    project.tasks = project.tasks.filter(
+      ({ taskId }) => task.taskId !== taskId,
+    )
+  }
+}
```

Comment:
Instead of `for` loop, consider:

```
programs.projects.forEach(project => {
  project.tasks.forEach((projectTask, projectTaskIndex) => {
    if (projectTask.taskId === task.taskId) {
      project.tasks[projectTaskIndex] = { ...task }
```